uaclient/config.py

```diff
@@ -104,7 +102,12 @@ class UAConfig:
         "apt_https_proxy",
     )
 
-    def __init__(self, cfg: Dict[str, Any] = None, series: str = None) -> None:
+    def __init__(
+        self,
+        cfg: Dict[str, Any] = None,
+        series: str = None,
+        root_mode: bool = True,
```

Comment:
Considering that it was always run as root before, we will have many instances where the root_mode is True and few with root_mode as False. So using the default as False, will make us change so many places.

Reply:
I think we still need to make that change. The rationale is that it is better for us to be explicit here. For example, Imagine in the future that we have the API structure ready. That means users would be able to run ua code directly on their application. On that situation, we need to have correct defaults, because we can assume that non-root users might be running that code.

Regarding the code changes, I think we would only need to add that logic when we create the config on the `main` function. And I don't see a problem with us updating the tests if needed.

Reply:
Okay. Makes sense, esp. on the future use.